### ISSUE TITLE
jStat multiply bug

### ISSUE THREAD

**Issue by gibito** [jstat/jstat#176]
Gents,

@trevnorris 
@mrwillihog 

Please shed some light here:

If you look at the console of this fiddle, https://jsfiddle.net/c8p5kv0r/3/

You will see that jStat calculations break at XTY variable in a very strange way:

When XTY is calculated as  XTY = jStat(XT).multiply(Y), we get a correct (compared to Excel calculations) 2 x 1 matrix BUT the following calculation using it, produces NaNs and a matrix with the wrong size (should be a 2 x 1)

![screen shot 2015-11-17 at 10 38 23 am](https://cloud.githubusercontent.com/assets/1581876/11205259/5fba28aa-8d17-11e5-9757-c85bb336b1b4.png)

When XTY is explicitly set (by uncommenting line var XTY = [[322], [6746]]; ), in console we get exactly the same XTY we got via jStat calculation BUT this time the following calculation of BETA_HAT is correct.

![screen shot 2015-11-17 at 10 42 00 am](https://cloud.githubusercontent.com/assets/1581876/11205326/e2848c8a-8d17-11e5-8e77-a9d968f84dc0.png)

Since XTY obtained by manual assignment and jStat calculation is identical, I have no idea why the following jStat calculation of BETA_HAT produces the wrong matrix size (it should be a 2x1 not a 2x2) and NaN. 

The only difference I spot is the "__proto__: Array[0]" vs "jStat[0]" in the XTY variable. 

Can this be the culprit of this bug?

Thanks.

**Comment by trevnorris** [comment]
Thanks for the test case. Will investigate why this is happening.

**Comment by trevnorris** [comment]
Minimal reproduction:
```js
jStat([[6, -2], [-2, 1]]).multiply(jStat([[322], [6746]]));
```

**Comment by trevnorris** [comment]
Believe this is linked to https://github.com/jstat/jstat/issues/55. The gist is there are a bunch of `isArray` checks only if the argument is an Array. If not then it takes a different path. So it should also check `instanceof jStat` and treat that as an array.

I'll work up a fix this weekend.

**Comment by trevnorris** [comment]
The fix has been released in 1.5.0.
